### PR TITLE
Dependency update

### PR DESCRIPTION
**Pull request by omnivion**
### Information
- JIRA story QPPCT-492.

### Changes proposed in this PR:
- Updates most of our dependencies

### Checklist 
- [X] All JUnit tests pass (`mvn clean verify`).
- [X] New unit tests written to cover new functionality.
- [X] Added and updated JavaDocs for non-test classes and methods.
- [X] No local design debt. Do you feel that something is "ugly" after your changes?
- [X] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: rest-api/pom.xml [hunk 1]
```diff
@@ -106,21 +106,21 @@
...
-			<version>0.1.4</version>
+			<version>0.1.7</version>
 			<scope>test</scope>
 		</dependency>
 
 		<dependency>
 			<groupId>org.springframework</groupId>
 			<artifactId>spring-test</artifactId>
-			<version>4.3.10.RELEASE</version>
+			<version>5.0.1.RELEASE</version>
```
Comment (halprin) [inline rest-api/pom.xml]: This is a big jump here.  Plus, SpringBoot version 1.5.8 is still on the Spring 4.* release.  I would keep this at the latest and greatest Spring 4.* release.

Comment (omnivion) [reply rest-api/pom.xml]: It's just the testing framework, and all tests pass.

Comment (sfradkin) [reply rest-api/pom.xml]: It doesn't try to bring in the Spring 5.0 libraries with this?

Comment (omnivion) [reply rest-api/pom.xml]: There's an <exclusions> block which excludes *.

**Pull request merged**